src/components/LeaderBoard/Leaderboard.jsx

```diff
@@ -160,6 +272,14 @@ const LeaderBoard = ({
           }}
         />
       </h3>
+      {(loggedInUser.role === 'Owner' || loggedInUser.role === 'Administrator' || loggedInUser.role == 'Core Team') && userTeams?.length > 0 && <button className='circle-border my-team' style={{ 
```

Comment:
we are doing this validation in different parts of the file, maybe it is worth to create a function to check this condition (if there is a utils file where this function can be added better) and create a variable to use where it is required.